src/phases/query/RepeatedPathTraversal.yml

```diff
@@ -0,0 +1,458 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: >
+  This workload stresses the query execution engine by running queries over a set of paths which
+  share a common prefix. Crucially, these queries never match a document in the collection.
+
+Keywords:
+- Loader
+- CrudActor
+- QuiesceActor
+- insert
+- find
+
+GlobalDefaults:
+  Database: &Database {^Parameter: {Name: Database, Default: unused}}
+  Repeat: &Repeat {^Parameter: {Name: Repeat, Default: -1}}
+  DocumentCount: &DocumentCount {^Parameter: {Name: DocumentCount, Default: -1}}
+  Collection: &Collection Collection0
+  Threads: &Threads 1
+  MaxPhases: &MaxPhases 20
+
+ActorTemplates:
+- TemplateName: FindQueryTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "unused"}}
+    Type: CrudActor
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: *Repeat
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter: {^Parameter: {Name: "Filter", Default: "invalid"}}
+
+Actors:
+# Clear any pre-existing collection state.
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+# Insert documents with 2 levels of nested arrays where the leaves contain many fields. This test
+# targets workloads where the paths queried share a common prefix ("a.b.c", "a.b.d", and so on).
+- Name: InsertRepeatedPathTraversalData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document: {
+          a: [{ ^Repeat: { count: 3, fromGenerator: { b: [&InnerDoc { c: 0, d: 0, e: 0, f: 0, g: 0, h: 0, i: 0, j: 0, k: 0, l: 0}]}}}],
+          # 5 levels deep
+          deep5: [
```

Comment:
As a note: this might generate a changepoint on the non-bonsai variants (build-barons don't care about the bonsai ones- worth flagging in the #perf-build-baron slack).